test/config/config.test.js

```diff
@@ -287,6 +286,141 @@ const testFixtures = [
       );
     },
   },
+  {
+    name: 'use agent - should parse values, log invalid ones and apply special cases',
+    script: 'fixtures/use-agent.js',
+    cwd: __dirname,
+    noConvenienceConfig: true,
+    env: {
+      ELASTIC_APM_TRANSACTION_MAX_SPANS: '-1',
+      ELASTIC_APM_BREAKDOWN_METRICS: 'false',
+      ELASTIC_APM_API_REQUEST_SIZE: '1mb',
+      ELASTIC_APM_ERROR_MESSAGE_MAX_LENGTH: '1mb',
+      ELASTIC_APM_ACTIVE: 'nope',
+      ELASTIC_APM_LOG_LEVEL: 'debug', // we want debug logs to check
+    },
+    checkScriptResult: (t, err, stdout) => {
+      t.error(err, `use-agent.js script succeeded: err=${err}`);
+      const reviver = (k, v) => (v === 'Infinity' ? Infinity : v);
+      const useAgentLogs = getUseAgentLogs(stdout);
+      const resolvedConfig = JSON.parse(useAgentLogs[2], reviver);
+      const warnLogs = getApmLogs(stdout, 'warn');
+      const debugLogs = getApmLogs(stdout, 'debug');
+
+      t.equal(
+        resolvedConfig.transactionMaxSpans,
+        Infinity,
+        'transactionMaxSpans can be set to Infinity (-1)',
+      );
+      t.equal(
+        resolvedConfig.breakdownMetrics,
+        false,
+        '"false" value for breakdownMetrics is parsed',
+      );
+      t.equal(
+        resolvedConfig.apiRequestSize,
+        1024 * 1024,
+        '"1mb" value is parsed for apiRequestSize',
+      );
+      t.equal(
+        warnLogs[0].message,
+        'config option "errorMessageMaxLength" is deprecated. Use "longFieldMaxLength"',
+        'got a warning about deprecated value',
+      );
+      t.equal(
+        resolvedConfig.errorMessageMaxLength,
+        1024 * 1024,
+        'bytes value is parsed for errorMessageMaxLength',
+      );
+      t.equal(
+        warnLogs[1].message,
+        'unrecognized boolean value "nope" for "active"',
+        'got a warning about bogus boolean value',
+      );
+      t.equal(
+        debugLogs[0].message,
+        'Elastic APM agent disabled (`active` is false)',
+        'got a debug log about agent disabled',
+      );
+      // XXX: normalization turns this value to `undefined` because "bogus"
```

Comment:
Note to reviewer: since `undefined` is falsy we may want to actually have `false` so we ensure we have the right type

Reply:
I think `undefined` should be treated the same as "not specified". What happens if the same config option is specified at multiple levels, e.g.: `ELASTIC_APM_ACTIVE=nope` and `apm.start({ active: true })`. Normally the envvar is a high priority, so it should win. But in this case, I think we'd expect to see:

- a warning that the `nope` value is invalid, and hence ignored
- active is `true` because the value from `apm.start(...)` was used.

I'm not sure if that answers your question. :)

Reply:
nit: Can we avoid commiting "XXX", or at least for very long, please? :) That's a string I use heavily as a marker for "this code is for local debugging/dev notes that is incomplete and shouldn't go in the release".

Reply:
sorry I left this on the code :(

I'll merge this one and create a new PR today moving more tests and removing this comment.